### PR TITLE
Multithreading

### PR DESCRIPTION
**Pull request by GopichandUlasalaEpam**
Task on MultiThreading is submitted.

### CODE REVIEW

File: MentorMultiThreading/MentorMultiThreading/Program.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            Console.WriteLine("Enter the size of the array");
+            int size =Convert.ToInt16(Console.ReadLine());
+            Console.WriteLine("Enter the array of integers (space-separated):");
+            string input = Console.ReadLine();
+            numberArray = Array.ConvertAll(input.Split(' '), int.Parse);
+            int midPoint = numberArray.Length / 2;
+            //First Thread
+            ThreadPool.QueueUserWorkItem(new WaitCallback(SumOperation), new LimitValues(0, midPoint));
+            //SecondThread
+            ThreadPool.QueueUserWorkItem(new WaitCallback(SumOperation), new LimitValues(midPoint, size));
```
Comment (csreehari) [inline MentorMultiThreading/MentorMultiThreading/Program.cs]:
```suggestion
            ThreadPool.QueueUserWorkItem(SumOperation, new LimitValues(0, midPoint));
            //SecondThread
            ThreadPool.QueueUserWorkItem(SumOperation, new LimitValues(midPoint, size));
```
could have simplified like above? See if this works. Same for below line as well.

File: MentorMultiThreading/MentorMultiThreading/Program.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            //making the main thread to wait
+            resetEvent.WaitOne();
+            Console.WriteLine("Main method thread");
+            int overrallEvenSum = firstSum;
+            Console.WriteLine($"From main thread the sum of even numbers of array is :{overrallEvenSum}");
+            
+        }
+        static void SumOperation(object values)
+        {
```
Comment (csreehari) [inline MentorMultiThreading/MentorMultiThreading/Program.cs:40]: As per task, you should handle any exceptions that may occur during the program's execution. There is a potential chance of exception in this case

File: MentorMultiThreading/MentorMultiThreading/Program.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            for (int i = start; i < end; i++)
+            {
+                if (numberArray[i] % 2 == 0)
+                {
+                    Interlocked.Add(ref firstSum, numberArray[i]);
+                }
+            }
+            if (start != 0)
+            {
+                resetEvent.Set();
```
Comment (csreehari) [inline MentorMultiThreading/MentorMultiThreading/Program.cs]: It is not clear as to which thread is completed here, but as per task: 

> You should use a ManualResetEvent object to signal when each thread has finished its work.

**Review by csreehari — CHANGES_REQUESTED** [review]

**Review by GopichandUlasalaEpam — COMMENTED** [review]
Refactored the made changes by introducing exception handling and making threads to start one after another using manualresetevent

**Pull request closed without merging**